SecretManagement.LastPass.psm1

```diff
@@ -0,0 +1,114 @@
+﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+using namespace Microsoft.PowerShell.SecretManagement
+
+$ModuleName = 'SecretManagement.LastPass'
+
+# The last segement (underscore + number (eg: _1)) is to view how many format args are expected.
+$ErrorMessages = @{
+    GetVaultParams0                 = "At least 1 vault implementing $ModuleName must be registered."
+    GetVaultParamsMany_1            = "`$Vault argument must be provided when multiple vault implementing $ModuleName exists: {0}"
+    StayConnectedForceSwitchMissing = 'StayConnected is a sensitive operation that save the LastPass decryption key on your hard drive. If you want to proceed, reissue the command specifying the force parameter.'
+    Unregister_NotLpass_1           = "The specified vault is not a $ModuleName vault (VaultType: {0}"
+}
+
+function Connect-LastPass {
+    [CmdletBinding()]
+    param (
+        [String]$Vault,
+        [String]$User,
+        [Switch]$Trust,
+        [Switch]$StayConnected,
+        [Switch]$Force 
+    )
+    $Arguments = [System.Collections.Generic.List[String]]@('login')
+    if ($trust) { $Arguments.Add('--trust') }
+    if ($StayConnected) { 
+        if (! $Force) { Throw [System.Management.Automation.PSArgumentException] $ErrorMessages.StayConnectedForceSwitchMissing }
+        $Arguments.Add('--plaintext-key') 
+        $Arguments.Add('--force')
+    }
+    $Arguments.Add($User)
+    $VaultParams = Get-VaultParams -Vault $Vault
+    Invoke-lpass -Arguments $Arguments -VaultParams $VaultParams
+}
+
+function Disconnect-LastPass {
+    [CmdletBinding()]
+    param (
+        [String]$Vault
+    )
+    $Arguments = [System.Collections.Generic.List[String]]@('logout', '--force')
+    $VaultParams = Get-VaultParams -Vault $Vault
+    Invoke-lpass -Arguments $Arguments -VaultParams $VaultParams
+}
+
+function Register-LastPassVault {
+    [CmdletBinding()]
+    param (
+        [String]$Vault,
+        [switch]$wsl,
+        [Switch]$Detailed,
+        [String]$Path
+    )
+
+    $Params = @{
+        ModuleName      = 'SecretManagement.LastPass'
+        Name            = if ('' -ne $Vault) {$Vault} else {$ModuleName}
+        VaultParameters = @{}
+    }
+    if ($wsl -eq $true) { $Params.VaultParameters.Add('wsl', $true) }
+    if ($Path -ne '') { $Params.VaultParameters.Add('lpassPath', $Path) }
+    if ($Detailed -eq $true) { $Params.VaultParameters.Add('outputType','Detailed') }
+    if ($VerbosePreference -eq 'Continue') {$Params.add('Verbose',$true)}
```

Comment:
```suggestion
    $Params = @{
        ModuleName      = 'SecretManagement.LastPass'
        Name            = if ('' -ne $Vault) {$Vault} else {$ModuleName}
        Verbose         = $VerbosePreference -eq 'Continue'
        VaultParameters = @{
            wsl         = $wsl.IsPresent
            outputType  = $Detailed.IsPresent
        }
    }

    if ($Path -ne '') { $Params.VaultParameters.Add('lpassPath', $Path) }
```

a tad cleaner IMO?

Reply:
I couldn't take the suggestion quite as it was presented because of outputType. 
It is not a boolean. 

```
$Params = @{
        ModuleName      = 'SecretManagement.LastPass'
        Name            = if ('' -ne $Vault) {$Vault} else {$ModuleName}
        Verbose         = $VerbosePreference -eq 'Continue'
        VaultParameters = @{
            wsl         = $Wsl.IsPresent
            outputType  = if ($Detailed) { 'Detailed' } else { 'Default' }
        }
    }
```

https://github.com/TylerLeonhardt/SecretManagement.LastPass/pull/18/commits/1d0107c5bc256437e32f77d56a49f9a0aeec9ae4